File: pdu/python/hako_srv_msgs/pdu_cdr_conv_JoinResponsePacket.py
from .pdu_pytype_JoinResponsePacket import JoinResponsePacket
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..hako_srv_msgs.pdu_cdr_conv_JoinResponse import *
from ..hako_srv_msgs.pdu_cdr_conv_ServiceResponseHeader import *



def py_to_cdr_body_JoinResponsePacket(writer: CdrWriter, src: JoinResponsePacket):
    py_to_cdr_body_ServiceResponseHeader(writer, src.header)
    py_to_cdr_body_JoinResponse(writer, src.body)


def cdr_body_to_py_JoinResponsePacket(reader: CdrReader, dst: JoinResponsePacket):
    cdr_body_to_py_ServiceResponseHeader(reader, dst.header)
    cdr_body_to_py_JoinResponse(reader, dst.body)
    return dst


def py_to_cdr_JoinResponsePacket(src: JoinResponsePacket) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_JoinResponsePacket(writer, src)
    return writer.bytes()


def cdr_to_py_JoinResponsePacket(cdr_payload) -> JoinResponsePacket:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = JoinResponsePacket()
    return cdr_body_to_py_JoinResponsePacket(reader, dst)

File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_JoinResponsePacket.js
import { JoinResponsePacket } from './pdu_jstype_JoinResponsePacket.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { JoinResponse } from '../hako_srv_msgs/pdu_jstype_JoinResponse.js';
import { PduJoinResponseConverter } from '../hako_srv_msgs/pdu_cdr_conv_JoinResponse.js';
import { ServiceResponseHeader } from '../hako_srv_msgs/pdu_jstype_ServiceResponseHeader.js';
import { PduServiceResponseHeaderConverter } from '../hako_srv_msgs/pdu_cdr_conv_ServiceResponseHeader.js';


export class PduJoinResponsePacketConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { JoinResponsePacket } src
     */
    static to_cdr_body(writer, src) {
        PduServiceResponseHeaderConverter.to_cdr_body(writer, src.header);
        PduJoinResponseConverter.to_cdr_body(writer, src.body);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { JoinResponsePacket } dst
     * @returns { JoinResponsePacket }
     */
    static cdr_body_to_js(reader, dst) {
        PduServiceResponseHeaderConverter.cdr_body_to_js(reader, dst.header);
        PduJoinResponseConverter.cdr_body_to_js(reader, dst.body);
        return dst;
    }

    /**
     * @param { JoinResponsePacket } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { JoinResponsePacket }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new JoinResponsePacket());
    }
}
